community/values/src/main/java/org/neo4j/values/storable/DateTimeValue.java

```diff
@@ -479,7 +478,7 @@ int unsafeCompareTo( Value other )
                     boolean thisIsOffset = thisZone instanceof ZoneOffset;
                     boolean thatIsOffset = thatZone instanceof ZoneOffset;
                     cmp = Boolean.compare( thatIsOffset, thisIsOffset ); // offsets before named zones
-                    if ( cmp == 0 && areDifferentZones( thisZone, thatZone ) )
+                    if ( cmp == 0 && (thisIsOffset || !areSameZone( thisZone, thatZone ) ) )
```

Comment:
I don't get why you are checking `thisIsOffset` explicitly here before call to `areSameZone`?

Reply:
When we get here, we know that either `this` and `that` are both Offsets or both Zones (otherwise `cmp` would be `!= 0`). If `thisIsOffset`, that means both are offsets. And then we can skip the call to `areSameZone`. which was only introduced to prevent Java from identifying renamed zones as equal.